File: Top Interview 150/12. Integer to Roman/12. Integer to Roman.js
var intToRoman = function(n) {
    let result = "";
    if (n>=1000){
        let m=(parseInt(n/1000));
        n-=parseInt(1000*parseInt((n/1000)));
        for (let i = 0;  i < m ; i ++ ){
            result+="M";
        }
    }
    if(n>=100){
        let c=(parseInt(n/100));
        n-=parseInt(100*parseInt((n/100)));
        if (c>=5){
            if(c===9){
                result+="CM";
            }
            else{
                result+="D";
                for(let i = 0 ; i < (c-5); i++){
                    result+="C";
                }
            }
        }
        else if (c===4){
            result+="CD";
        }
        else{
            for (let i = 0 ; i < c; i++){
                result+="C";
            }
        }
    }
    if(n>=10){
        let d=(parseInt(n/10));
        n-=parseInt(10*parseInt((n/10)));
        
        if (d>=5){
            if(d===9){
                result+="XC";
            }
            else {
                result+="L";
                for(let i = 0 ; i < d-5; i++){
                    result+="X";
                }
            } 
        }
        else if (d===4){
            result+="XL";
        }
        else{
            for(let i = 0 ; i < d; i++){
                result+="X";
            }
        }
    }
    if (n>=1){
        let u=parseInt(n);
        if (u>=5){
            if(u===9){
                result+="IX"
            }
            else{
                result+="V";
                for(let i = 0; i < u-5;i++){
                    result+="I";
                }
            }
        }
        else if (u===4){
            result+="IV";
        }
        else{
            for(let i = 0; i < u;i++){
                result+="I";
            }
        }
    }
    return result;
};

File: Top Interview 150/12. Integer to Roman/12. Integer to Roman.py
class Solution:
    def intToRoman(self, num: int) -> str:
        result=""
        
        if num>=1000:
            m="M"*(num//1000)
            num-=1000*(num//1000)
            result+=m
        if num >= 100:
            if num//100==9:
                c="CM"
            elif num//100>=5:
                c="D"+("C"*((num//100)-5))
            elif num//100==4:
                c="CD"
            else :
                c=("C"*(num//100))
            num-=100*(num//100)
            result+=c
        if num >= 10:
            if num//10==9:
                d="XC"
            elif num//10>=5:
                d="L"+("X"*((num//10)-5))
            elif num//10==4:
                d="XL"
            else:
                d="X"*(num//10)
            num-=10*(num//10)
            result+=d
        if num >= 1:
            if num==9:
                u="IX"
            elif num>=5:
                u="V"+("I"*((num)-5))
            elif num==4:
                u="IV"
            else:
                u="I"*(num)
            result+=u
        return result